src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -141,45 +124,59 @@ public final Object runTool(){
     }
 
     public Object instanceMainPostParseArgs() {
-        // Provide one temp directory if the caller didn't
-        if (this.TMP_DIR == null) this.TMP_DIR = new ArrayList<>();
-        if (this.TMP_DIR.isEmpty()) TMP_DIR.add(IOUtil.getDefaultTmpDir());
-
         // Build the default headers
         final ZonedDateTime startDateTime = ZonedDateTime.now();
         this.defaultHeaders.add(new StringHeader(commandLine));
         this.defaultHeaders.add(new StringHeader("Started on: " + Utils.getDateTimeForDisplay(startDateTime)));
 
-        LoggingUtils.setLoggingLevel(VERBOSITY);  // propagate the VERBOSITY level to logging frameworks
+        LoggingUtils.setLoggingLevel(configArgs.getVerbosity());  // propagate the VERBOSITY level to logging frameworks
 
-        for (final File f : TMP_DIR) {
+        for (final Path p : this.configArgs.getTmpDirectories()) {
             // Intentionally not checking the return values, because it may be that the program does not
             // need a tmp_dir. If this fails, the problem will be discovered downstream.
-            if (!f.exists()) f.mkdirs();
-            f.setReadable(true, false);
-            f.setWritable(true, false);
-            System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
+            if (!Files.exists(p)) {
+                try {
+                    Files.createDirectories(p);
+                } catch (IOException e) {
+                    // intentionally ignoring
+                }
+            }
+            try {
+                // TODO - not sure if this will work as before, because using File.setReadable(true, false) and
+                // TODO - f.setWritable(true, false) was done in two steps, so if one failed the other was done
+                // TODO - I don't know if the setPosixFilePermissions try to set all and then throws an exception
+                // TODO - if any of them fails, or just has an early termination
+                // TODO - an option to avoid this problem is iterate over a list of the permissions and set them
+                // TODO - one by one, ignoring possible errors (this was done in the original commit/PR)
+                Files.setPosixFilePermissions(p, new HashSet<>(Arrays.asList(
+                        PosixFilePermission.OWNER_READ, PosixFilePermission.GROUP_READ, PosixFilePermission.OTHERS_READ,
+                        PosixFilePermission.OWNER_WRITE, PosixFilePermission.GROUP_WRITE, PosixFilePermission.OTHERS_WRITE)
+                ));
+            } catch (UnsupportedOperationException | IOException e) {
+                // intentionally ignoring
+            }
+            System.setProperty("java.io.tmpdir", p.toAbsolutePath().toUri().toString()); // in loop so that last one takes effect
```

Comment:
It looks like this line is causing the failure, because `java.io.tmpdir` is using the full URI even for the default FS. In my local computer it appears a folder called `file://` with the sub-folder structure following the `$TMPDIR` variable set in my system.

I would try locally if the `p.toAbsolutePath().toString()` works in this case, but I am not sure if this will work for `java.nio.Path` in other filesystems. Any idea @droazen and/or @lbergelson?